File: android/app/src/main/java/dev/cloveclove/peppercheck/data/evidence/TaskEvidence.kt
package dev.cloveclove.peppercheck.data.evidence

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class TaskEvidence(
    val id: String,
    @SerialName("task_id") 
    val taskId: String,
    val description: String,
    val status: String, // pending_upload, ready
    @SerialName("created_at") 
    val createdAt: String,
    @SerialName("updated_at") 
    val updatedAt: String,
    @SerialName("task_evidence_assets") 
    val taskEvidenceAssets: List<TaskEvidenceAsset> = emptyList()
)

@Serializable
data class TaskEvidenceAsset(
    val id: String,
    @SerialName("evidence_id") 
    val evidenceId: String,
    @SerialName("file_url") 
    val fileUrl: String,
    @SerialName("file_size_bytes") 
    val fileSizeBytes: Long? = null,
    @SerialName("content_type") 
    val contentType: String? = null,
    @SerialName("created_at") 
    val createdAt: String,
    @SerialName("public_url") 
    val publicUrl: String? = null,
    @SerialName("processing_status") 
    val processingStatus: String? = null,
    @SerialName("error_message") 
    val errorMessage: String? = null
)

File: peppercheck_flutter/lib/features/evidence/domain/task_evidence.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'task_evidence.freezed.dart';
part 'task_evidence.g.dart';

// ignore_for_file: invalid_annotation_target

@freezed
abstract class TaskEvidence with _$TaskEvidence {
  const factory TaskEvidence({
    required String id,
    @JsonKey(name: 'task_id') required String taskId,
    required String description,
    @Default('pending_upload') String status,
    @JsonKey(name: 'created_at') required String createdAt,
    @JsonKey(name: 'updated_at') required String updatedAt,
    @JsonKey(name: 'task_evidence_assets')
    @Default([])
    List<TaskEvidenceAsset> assets,
  }) = _TaskEvidence;

  factory TaskEvidence.fromJson(Map<String, dynamic> json) =>
      _$TaskEvidenceFromJson(json);
}

@freezed
abstract class TaskEvidenceAsset with _$TaskEvidenceAsset {
  const factory TaskEvidenceAsset({
    required String id,
    @JsonKey(name: 'evidence_id') required String evidenceId,
    @JsonKey(name: 'file_url') required String fileUrl,
    @JsonKey(name: 'file_size_bytes') int? fileSizeBytes,
    @JsonKey(name: 'content_type') String? contentType,
    @JsonKey(name: 'public_url') String? publicUrl,
    @JsonKey(name: 'processing_status')
    @Default('pending')
    String processingStatus,
    @JsonKey(name: 'error_message') String? errorMessage,
    @JsonKey(name: 'created_at') required String createdAt,
  }) = _TaskEvidenceAsset;

  factory TaskEvidenceAsset.fromJson(Map<String, dynamic> json) =>
      _$TaskEvidenceAssetFromJson(json);
}
